xformers/components/attention/__init__.py

```diff
@@ -77,10 +78,11 @@ class MyAttention(Attention):
 )
 
 
-def maybe_sparsify(matrix):
+def maybe_sparsify(matrix) -> Any:
     # Sparsify if that makes sense
     if torch.count_nonzero(matrix).item() / matrix.numel() > _DENSITY_THRESHOLD:
-        return matrix
+        # If not sparse, then AttentionMask is the reference type
```

Comment:
heads up on that, basically hijacking any non-sparse case to put it under the AttentionMask umbrella